labelbox/client.py

```diff
@@ -939,3 +940,306 @@ def get_model_run(self, model_run_id: str) -> ModelRun:
+                        assignments=res['sanitizedAssignments'],
+                        sanitized=True))
+                results.extend(
+                    _format_successful_assignments(
+                        assignments=res['unmodifiedAssignments'],
+                        sanitized=False))
+                # Failed assignments
+                errors.extend(
+                    _format_failed_assignments(
+                        assignments=res['invalidGlobalKeyAssignments'],
+                        error_msg="Invalid global key"))
+                errors.extend(
+                    _format_failed_assignments(
+                        assignments=res['accessDeniedAssignments'],
+                        error_msg="Access denied to Data Row"))
+
+                if len(errors) == 0:
+                    status = "Success"
+                elif len(results) > 0:
+                    status = "Partial Success"
+                else:
+                    status = "Failure"
+
+                if len(errors) > 0:
+                    logger.warning(
+                        "There are errors present. Please look at 'errors' in the returned dict for more details"
+                    )
+
+                return {
+                    "status": status,
+                    "results": results,
+                    "errors": errors,
+                }
+            elif res["assignGlobalKeysToDataRowsResult"][
+                    "jobStatus"] == "FAILED":
+                raise labelbox.exceptions.LabelboxError(
+                    "Job assign_global_keys_to_data_rows failed.")
+            current_time = time.time()
+            if current_time - start_time > timeout_seconds:
+                raise labelbox.exceptions.TimeoutError(
+                    "Timed out waiting for assign_global_keys_to_data_rows job to complete."
+                )
+            time.sleep(sleep_time)
+
+    def get_data_row_ids_for_global_keys(
+            self,
+            global_keys: List[str],
+            timeout_seconds=60) -> List[Dict[str, str]]:
+        """
+        Gets data row ids for a list of global keys.
+
+        Args:
+            A list of global keys
+        Returns:
+            Dictionary containing 'status', 'results' and 'errors'.
+
+            'Status' contains the outcome of this job. It can be one of
+            'Success', 'Partial Success', or 'Failure'.
+
+            'Results' contains a list of data row ids successfully fetchced. It may
+            not necessarily contain all data rows requested.
+
+            'Errors' contains a list of global_keys that could not be fetched, along
+            with the failure reason
+        Examples:
+            >>> job_result = client.get_data_row_ids_for_global_keys(["key1","key2"])
+            >>> print(job_result['status'])
+            Partial Success
+            >>> print(job_result['results'])
+            ['cl7tv9wry00hlka6gai588ozv', 'cl7tv9wxg00hpka6gf8sh81bj']
```

Comment:
**question** the more i think about this, the more i wonder if this should be a dict with global key and data row? i know Product said that this would be okay, but if i am user, is the expectation that i should know the direct mapping of datarow <> global key all the time after assigning it once?